### PR TITLE
chore(NODE-6319): update dev dependencies and add dependabot config

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

Dev dependencies are updated.  Dependabot configuration has been added.

CI fails with the same failures as `main`, nothing new in this PR.

### Double check the following

- [ ] Ran `npm run check:eslint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -77,11 +76,10 @@
         "null": "ignore"
       }
     ],
+    "@typescript-eslint/no-unused-expressions": "off",
+    "@typescript-eslint/no-require-imports": "off",
     "simple-import-sort/imports": "error",
     "simple-import-sort/exports": "error",
```
Comment (aditi-khare-mongoDB) [inline .eslintrc.json:82]: We want to add in the `eslint/no-duplicate-imports`, right? (linked [here](https://eslint.org/docs/latest/rules/no-duplicate-imports))

**Review by aditi-khare-mongoDB — CHANGES_REQUESTED** [review]

**Review by aditi-khare-mongoDB — APPROVED** [review]
Waiting for our conversation on virtual office to resolve before merging - so we can add in other rules if needed

File: package.json [hunk 2]
```diff
@@ -7,27 +7,26 @@
   ],
   "devDependencies": {
```
Comment (aditi-khare-mongoDB) [inline package.json:8]: Some tests are failing, any idea why?

**Review by aditi-khare-mongoDB — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (baileympearson) [reply package.json:8]: The tests are also failing on `main`.

**Review by aditi-khare-mongoDB — APPROVED** [review]

**Pull request merged**